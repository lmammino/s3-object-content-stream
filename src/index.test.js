const S3ObjectContentStream = require(process.env.BUILD
  ? '../pkg/dist-node/'
  : './')

class MockS3 {
  constructor (mockedPages, shouldFail = false) {
    this._mockedPages = mockedPages
    this._currentPage = 0
    this._shouldFail = shouldFail
    this._receivedParams = []
  }

  listObjectsV2 (params, cb) {
    this._receivedParams.push(params)
    process.nextTick(() => {
      const error = this._shouldFail ? new Error('some error') : null
      const response = this._shouldFail
        ? null
        : this._mockedPages[this._currentPage++]
      cb(error, response)
    })
  }
}

const createPage = (numItems, startCount = 0, last = false) => {
  const items = []
  for (let i = 0; i < numItems; i++) {
    items.push({
      ETag: `etag-${i + startCount}`,
      Key: `${i + startCount}.jpg`,
      LastModified: '2019-02-19T19:35:20.892Z',
      Size: (i + startCount) * 100,
      StorageClass: 'STANDARD'
    })
  }

  return {
    Contents: items,
    IsTruncated: !last,
    KeyCount: numItems,
    MaxKeys: numItems,
    Name: 'examplebucket',
    NextContinuationToken: last ? undefined : `token-${numItems + startCount}`,
    Prefix: ''
  }
}
