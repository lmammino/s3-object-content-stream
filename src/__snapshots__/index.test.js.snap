// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should be possible to use a transform stream to transform the content of every single object 1`] = `
Array [
  Object {
    "Bucket": "some-bucket",
    "Key": "files/file1",
  },
  Object {
    "Bucket": "some-bucket",
    "Key": "files/file2",
  },
  Object {
    "Bucket": "some-bucket",
    "Key": "files/file3",
  },
]
`;

exports[`It should operate if the source stream is in fullMetadata mode 1`] = `
Array [
  Object {
    "Bucket": "some-bucket",
    "Key": "files/file1",
  },
  Object {
    "Bucket": "some-bucket",
    "Key": "files/file2",
  },
  Object {
    "Bucket": "some-bucket",
    "Key": "files/file3",
  },
]
`;

exports[`It should return concatenated output from multiple buckets 1`] = `
Array [
  Object {
    "Bucket": "some-bucket",
    "Key": "files/file1",
  },
  Object {
    "Bucket": "some-bucket",
    "Key": "files/file2",
  },
  Object {
    "Bucket": "some-bucket",
    "Key": "files/file3",
  },
]
`;
